module/VuFindSearch/src/VuFindSearch/Backend/WorldCat2/Response/RecordCollectionFactory.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Simple factory for WorldCat v2 record collection.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   David Maus <maus@hab.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+
+namespace VuFindSearch\Backend\WorldCat2\Response;
+
+use VuFindSearch\Exception\InvalidArgumentException;
+use VuFindSearch\Response\RecordCollectionFactoryInterface;
+
+use function call_user_func;
+use function gettype;
+use function is_array;
+use function is_callable;
+use function sprintf;
+
+/**
+ * Simple factory for WorldCat v2 record collection.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   David Maus <maus@hab.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class RecordCollectionFactory implements RecordCollectionFactoryInterface
+{
+    /**
+     * Factory to turn data into a record object.
+     *
+     * @var callable
+     */
+    protected $recordFactory;
+
+    /**
+     * Class of collection.
+     *
+     * @var string
+     */
+    protected $collectionClass;
+
+    /**
+     * Constructor.
+     *
+     * @param callable $recordFactory   Record factory function (null for default)
+     * @param string   $collectionClass Class of collection
+     *
+     * @return void
+     */
+    public function __construct($recordFactory = null, $collectionClass = null)
+    {
+        if (null === $recordFactory) {
+            $recordFactory = function ($data) {
+                return new Record($data);
+            };
+        } elseif (!is_callable($recordFactory)) {
+            throw new InvalidArgumentException('Record factory must be callable.');
+        }
+        $this->recordFactory = $recordFactory;
+        $this->collectionClass = $collectionClass ?? RecordCollection::class;
+    }
+
+    /**
+     * Return record collection.
+     *
+     * @param array $response Collection of XML documents
```

Comment:
Haven't done the review yet but coincidentally someone just asked about whether this new API could return XML, and this line hints yes, but I think it's a lie because https://github.com/vufind-org/vufind/pull/2612/files#diff-d8b51aac0547dd71cd90709b806fd885c5ea954bec687dd59da4e90eeda53453R50